jena-core/src/main/java/org/apache/jena/graph/Factory.java

```diff
@@ -40,11 +40,11 @@ public static Graph createDefaultGraph()
         { return Factory.createGraphMem( ); }
 
     public static Graph createGraphMem()
-        { return new GraphMem(); }
+        { return new GraphMemUsingHashMap(); }
 
     public static Graph createGraphMemWithTransactionHandler( final TransactionHandler th )
         {
-        Graph g = new GraphMem()
+        Graph g = new GraphMemUsingHashMap()
```

Comment:
I'm not sure we want to be just blindly changing the default implementation out from underneath users like this.  If memory serves when we've done this kind of change in the past we've provided means to use the new implementation with a recommendation that users start testing with the new implementation first, and then changed the default later.

This allows time for any gotchas, pitfalls, bugs etc to be discovered and worked out rather than taking users by surprise.

For example you point out in one of the adjusted test cases that there's an assumption that the existing implementation iterates triples in a particular order.  Even if this isn't formally specified in the contract anywhere when an implementation has been in use for so long there's bound to be users relying on that behaviour (whether intentionally or otherwise)

Reply:
I don't see a way to do this because application code is mostly calling "createDefaultModel" so it is oblivious to the implementation choice. There are some deprecations in 4.4.0 for code directly using GraphMem (which it shouldn't be!).

Reply:
See also discussion: https://github.com/apache/jena/pull/1273#issuecomment-1112634490

I'll mark this resolved and lets' continue the migration discussion on the overall thread.